pkg/controller/compliancescan/compliancescan_controller_test.go

```diff
@@ -425,16 +430,24 @@ var _ = Describe("Testing compliancescan controller phases", func() {
 
 	Context("On the DONE phase", func() {
 		Context("with delete flag off", func() {
+			var secret corev1.Secret
+			var err error
 			BeforeEach(func() {
 				// Create the pods and the secret for the test
 				createFakeScanPods(reconciler, compliancescaninstance.Name, nodeinstance1.Name, nodeinstance2.Name)
-				createFakeRsSecret(reconciler, compliancescaninstance.Name)
+				secret, err = createFakeRsSecret(reconciler, compliancescaninstance.Name)
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
This is failing because the error returned is a 409, even though we expect `createFakeRsSecret` to create the secret we want to test with...

Seems like something isn't cleaning up properly?